libs/cardano-ledger-test/bench/Bench/Cardano/Ledger/StakeDistr.hs

```diff
@@ -0,0 +1,388 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- | Benchmark the TickF rule. and its major sub-computations and rules, to discover where the time is spent
+module Bench.Cardano.Ledger.StakeDistr
+  ( tickfRuleBench,
+  )
+where
+
+import Cardano.Binary (FromCBOR (..), ToCBOR (..), decodeFullDecoder)
+import Cardano.Ledger.Alonzo (AlonzoEra)
+import Cardano.Ledger.Alonzo.PParams (AlonzoPParamsHKD (..))
+import Cardano.Ledger.BaseTypes (BlocksMade (..), Globals (..), pvMajor)
+import Cardano.Ledger.Coin (DeltaCoin (..))
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Crypto (StandardCrypto)
+import Cardano.Ledger.Era (EraCrypto)
+import Cardano.Ledger.PoolDistr (PoolDistr (..))
+import Cardano.Ledger.Shelley.EpochBoundary (SnapShot (..), SnapShots (..))
+import Cardano.Ledger.Shelley.Genesis (ShelleyGenesis (..), mkShelleyGlobals)
+import Cardano.Ledger.Shelley.LedgerState
+  ( EpochState (..),
+    Filtered (..),
+    NewEpochState (..),
+    PulsingRewUpdate (Complete, Pulsing),
+    RewardUpdate (..),
+    applyRUpd',
+    dpsDState,
+    filterAllRewards,
+    lsDPState,
+    rewards,
+  )
+import Cardano.Ledger.Shelley.Rewards (aggregateRewards, sumRewards)
+import Cardano.Ledger.Shelley.Rules
+  ( ShelleyEPOCH,
+    ShelleyMIR,
+    ShelleyNEWEPOCH,
+    ShelleyTICKF,
+    adoptGenesisDelegs,
+    calculatePoolDistr,
+    calculatePoolStake,
+    updateRewards,
+    validatingTickTransition,
+  )
+import Cardano.Ledger.Slot (EpochNo, SlotNo (..))
+import Cardano.Slotting.EpochInfo (fixedEpochInfo)
+import Cardano.Slotting.Slot (EpochNo (..), SlotNo)
+import Cardano.Slotting.Time (mkSlotLength)
+import Control.DeepSeq (NFData (rnf))
+import Control.Exception (try)
+import Control.Monad.Reader (runReader)
+import Control.State.Transition.Extended (TRC (TRC), applySTS, runRule)
+import Criterion (Benchmark, bench, bgroup, env, whnf)
+import qualified Data.Aeson as Aeson (decode)
+import Data.ByteString.Lazy as Lazy (readFile)
+import Data.Coders
+import Data.Default.Class (Default (def))
+import qualified Data.Map.Strict as Map
+import Data.Maybe (fromJust)
+import Data.Maybe.Strict (StrictMaybe (..))
+import Data.Strings (lazyBytes)
+import Data.Text (pack)
+import qualified Data.UMap as UM
+import GHC.IO.Exception (IOException (..))
+import Test.Cardano.Ledger.Tickf (oldCalculatePoolDistr)
+
+-- ======================================================================
+
+type CurrentEra = AlonzoEra StandardCrypto
```

Comment:
My choice of Alonzo was indicated because the slot that we got the NewEpochState from was in the Alonzo era.
Because the code we benchmark runs the same in every era, it probably doesn't matter.